File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/AgeGroup.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum AgeGroup {
    PRESCHOOL("Okul Öncesi"), // 0-7
    ELEMENTARY("İlköğretim Çağı"), // 7-14
    ALL_AGES("Her Yaş");

    private final String displayName;

    AgeGroup(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: frontend/lib/models/enum/age_group.dart
enum AgeGroup {
  preschool(
    'Okul Öncesi',
    'PRESCHOOL',
  ),
  elementary('İlköğretim Çağı', 'ELEMENTARY'), //7- 14
  allAges('Her Yaş', 'ALL_AGES');

  final String displayName;
  final String apiValue;

  const AgeGroup(this.displayName, this.apiValue);

  static AgeGroup? fromApiValue(String? value) {
    if (value == null) return null;
    return AgeGroup.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen yaş aralığı: $value'),
    );
  }
}
